tests/test_plots.py

```diff
@@ -1,238 +1,318 @@
+@pytest.mark.parametrize('phase', [True, False])
+def test_spherical_phase(function, phase):
+    """Test all spherical plots with custom arguments."""
+    coords = sp.samplings.equal_area(n_max=0, n_points=500)
+    data = np.sin(coords.colatitude) * np.cos(coords.azimuth)
+    if phase:
+        # use complex data to test phase plotting
+        data = np.exp(1j * data)
+
+    # do plotting
+    filename = f'phase_{function.__name__}_{phase}'
+    create_figure()
+    function(coords, data, phase=phase)
+    save_and_compare(create_baseline, baseline_path, output_path, filename,
+                     file_type, compare_output)
+
+    # test error for invalid inputs
+    match = 'phase must be a boolean.'
+    with pytest.raises(ValueError, match=match):
+        function(coords, data, phase=5)
+
+
+@pytest.mark.parametrize('function', [
+    (sp.plot.pcolor_map),
+    (sp.plot.contour_map),
+    ])
+@pytest.mark.parametrize('projection', ['mollweide', 'hammer'])
+def test_spherical_projection(function, projection):
+    """Test all spherical plots with custom arguments."""
+    coords = sp.samplings.equal_area(n_max=0, n_points=500)
+    data = np.sin(coords.colatitude) * np.cos(coords.azimuth)
+
+    # do plotting
+    filename = f'projection_{function.__name__}_{projection}'
+    create_figure()
+    function(coords, data, projection=projection)
+    save_and_compare(create_baseline, baseline_path, output_path, filename,
+                     file_type, compare_output)
+
+    # test error for invalid inputs
+    match = "The projection of the axis needs to be 'projection'"
+    with pytest.raises(ValueError, match=match):
+        function(coords, data, projection='projection')
+
+
+@pytest.mark.parametrize('function', [
+    sp.plot.contour_map,
+    # placeholder after level implementation in sp.plot.contour,
+    ])
+@pytest.mark.parametrize('levels', [
+    (-0.5, .5),
+    [-0.5, .5],
+    np.array((-0.5, .5)),
+    ])
+def test_spherical_levels(function, levels):
+    """Test all spherical plots with custom arguments."""
+    coords = sp.samplings.equal_area(n_max=0, n_points=500)
+    data = np.sin(coords.colatitude) * np.cos(coords.azimuth)
+
+    # do plotting
+    filename = f'levels_{function.__name__}_{type(levels).__name__}'
+    create_figure()
+    function(coords, data, levels=levels)
+    save_and_compare(create_baseline, baseline_path, output_path, filename,
+                     file_type, compare_output)
+
+    # test error for invalid inputs
+    match = 'levels'
+    with pytest.raises(ValueError, match=match):
+        function(coords, data, levels='levels')
+
+
+@pytest.mark.parametrize(("function", "projection"), [
+    (sp.plot.balloon, '3d'),
+    (sp.plot.balloon_wireframe, '3d'),
+    (sp.plot.pcolor_sphere, '3d'),
+    (sp.plot.contour, 'rectilinear'),
+    (sp.plot.contour_map, 'mollweide'),
+    (sp.plot.pcolor_map, 'mollweide'),
+    ])
+def test_axes_input_and_return(function, projection):
```

Comment:
```suggestion
def test_projection_input_and_return(function, projection):
```
Maybe more fitting here.